Flaky-test: BrokerDispatchRateLimiterTest.testBrokerDispatchThrottledMetrics

### Search before reporting

- [x] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/apache/pulsar/actions/runs/20657650595/job/59313928001?pr=25104#step:11:1019

### Exception stacktrace

```text
Error:  Tests run: 5, Failures: 1, Errors: 0, Skipped: 4, Time elapsed: 27.86 s <<< FAILURE! -- in org.apache.pulsar.broker.service.BrokerDispatchRateLimiterTest
  Error:  org.apache.pulsar.broker.service.BrokerDispatchRateLimiterTest.testBrokerDispatchThrottledMetrics -- Time elapsed: 9.457 s <<< FAILURE!
  java.io.IOException: java.util.concurrent.ExecutionException: org.mockito.exceptions.misusing.DisabledMockException: Mock accessed after inline mocks were cleared
  	at org.apache.pulsar.PrometheusMetricsTestUtil.generate(PrometheusMetricsTestUtil.java:63)
  	at org.apache.pulsar.PrometheusMetricsTestUtil.generate(PrometheusMetricsTestUtil.java:40)
  	at org.apache.pulsar.broker.service.BrokerDispatchRateLimiterTest.testBrokerDispatchThrottledMetrics(BrokerDispatchRateLimiterTest.java:113)
  	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
  	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
  	at org.testng.internal.invokers.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:76)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
  	at java.base/java.lang.Thread.run(Thread.java:1583)
  Caused by: java.util.concurrent.ExecutionException: org.mockito.exceptions.misusing.DisabledMockException: Mock accessed after inline mocks were cleared
  	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
  	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
  	at org.apache.pulsar.PrometheusMetricsTestUtil.generate(PrometheusMetricsTestUtil.java:58)
  	... 12 more
  Caused by: org.mockito.exceptions.misusing.DisabledMockException: Mock accessed after inline mocks were cleared
  	at org.apache.pulsar.broker.PulsarService.getConfig(PulsarService.java:215)
  	at org.apache.pulsar.broker.service.BrokerService.getPendingLookupRequest(BrokerService.java:513)
  	at org.apache.pulsar.broker.stats.prometheus.metrics.ObserverGauge.collect(ObserverGauge.java:96)
  	at io.prometheus.client.Collector.collect(Collector.java:45)
  	at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.findNextElement(CollectorRegistry.java:204)
  	at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.nextElement(CollectorRegistry.java:219)
  	at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.nextElement(CollectorRegistry.java:152)
  	at org.apache.pulsar.broker.stats.prometheus.PrometheusMetricsGeneratorUtils.generateSystemMetrics(PrometheusMetricsGeneratorUtils.java:60)
  	at org.apache.pulsar.broker.stats.prometheus.PrometheusMetricsGenerator.generateMetrics(PrometheusMetricsGenerator.java:326)
  	at org.apache.pulsar.broker.stats.prometheus.PrometheusMetricsGenerator.lambda$renderToBuffer$0(PrometheusMetricsGenerator.java:538)
  	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
  	at org.apache.pulsar.broker.stats.prometheus.PrometheusMetricsGenerator.renderToBuffer(PrometheusMetricsGenerator.java:536)
  	at org.apache.pulsar.PrometheusMetricsTestUtil.generate(PrometheusMetricsTestUtil.java:54)
  	... 12 more
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BrokerDispatchRateLimiterTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BrokerDispatchRateLimiterTest.java
@@ -20,11 +20,14 @@ package org.apache.pulsar.broker.service;
 
 import static org.apache.pulsar.broker.stats.prometheus.PrometheusMetricsClient.parseMetrics;
 import static org.testng.Assert.assertEquals;
+import static org.testng.AssertJUnit.assertFalse;
 import com.google.common.collect.Multimap;
 import java.io.ByteArrayOutputStream;
+import java.time.Duration;
 import java.util.Collection;
 import java.util.UUID;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
 import lombok.Cleanup;
 import org.apache.pulsar.PrometheusMetricsTestUtil;
 import org.apache.pulsar.broker.stats.prometheus.PrometheusMetricsClient;
@@ -35,6 +38,7 @@ import org.apache.pulsar.client.api.Producer;
 import org.apache.pulsar.client.api.Schema;
 import org.apache.pulsar.client.api.SubscriptionType;
 import org.awaitility.Awaitility;
+import org.mockito.exceptions.misusing.DisabledMockException;
 import org.testng.Assert;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
@@ -109,10 +113,18 @@ public class BrokerDispatchRateLimiterTest extends BrokerTestBase {
         }
 
         // Assert broker metrics
-        ByteArrayOutputStream output = new ByteArrayOutputStream();
-        PrometheusMetricsTestUtil.generate(pulsar, true, false, false, output);
-        String metricsStr = output.toString();
-        Multimap<String, PrometheusMetricsClient.Metric> metrics = parseMetrics(metricsStr);
+        AtomicReference<String> metricsString = new AtomicReference<>();
+        Awaitility.await()
+                .pollInterval(Duration.ofMillis(200))
+                .atMost(Duration.ofSeconds(5))
+                .ignoreExceptionsMatching(exception -> exception.getCause() instanceof DisabledMockException)
+                .untilAsserted(() -> {
+                    ByteArrayOutputStream output = new ByteArrayOutputStream();
+                    PrometheusMetricsTestUtil.generate(pulsar, true, false, false, output);
+                    metricsString.set(output.toString());
+                    assertFalse(metricsString.get().isEmpty());
+                });
+        Multimap<String, PrometheusMetricsClient.Metric> metrics = parseMetrics(metricsString.get());
 
         // Assert subscription metrics reason by broker limit
         Collection<PrometheusMetricsClient.Metric> subscriptionDispatchThrottledMsgCountMetrics =

